File: app/src/main/java/com/dominikgaller/alpinebooking/booking/core/domain/event/TourBookingConfirmed.java
package com.dominikgaller.alpinebooking.booking.core.domain.event;

import com.dominikgaller.alpinebooking.shared.domain.event.DomainEvent;

import com.dominikgaller.alpinebooking.booking.core.domain.BookingId;

import java.time.Instant;

/**
 * Domain event emitted when a tour booking is confirmed.
 *
 * <p>Immutable. Published after successful transaction commit (see ADR 0002).
 *
 * <p>SDD: See {@code documentation/use-cases/uc02-confirm-tour-booking.spec.md}, section 7.
 */
public record TourBookingConfirmed(
        BookingId bookingId,
        Instant occurredAt
) implements DomainEvent {
}

File: src/main/kotlin/com/example/service/booking/core/domain/tourbooking/event/TourBookingConfirmed.kt
package com.example.service.booking.core.domain.tourbooking.event

import com.example.service.booking.core.domain.tourbooking.BookingId
import com.example.service.shared.domain.event.DomainEvent
import java.time.Instant

/**
 * A requested booking was confirmed (UC02).
 *
 * SDD: see `documentation/use-cases/uc02-confirm-tour-booking.spec.md` § 6.
 */
data class TourBookingConfirmed(
    val bookingId: BookingId,
    val occurredAt: Instant,
) : DomainEvent
